sqlmesh/core/analytics/__init__.py

```diff
@@ -31,9 +31,13 @@ def disable_analytics() -> None:
     collector = AnalyticsCollector(dispatcher=NoopEventDispatcher())
 
 
-def cli_analytics(func: t.Callable[..., t.Any]) -> t.Callable[..., t.Any]:
+P = t.ParamSpec("P")
+T = t.TypeVar("T")
+
+
+def cli_analytics(func: t.Callable[P, T]) -> t.Callable[P, T]:
```

Comment:
It looks like there's an [alternative form](https://docs.python.org/3/library/typing.html#typing.ParamSpec) we may be able to use here:

```python
def add_logging[T, **P](f: Callable[P, T]) -> Callable[P, T]:
    '''A type-safe decorator to add logging to a function.'''
    def inner(*args: P.args, **kwargs: P.kwargs) -> T:
        logging.info(f'{f.__name__} was called')
        return f(*args, **kwargs)
    return inner
```

Haven't encountered this before, but perhaps we can get rid of the `P`, `T` definitions in favor of the more compact version? Ditto for `python_api_analytics`.

Reply:
none of these changes seem relevant, we should revert

Reply:
> It looks like there's an [alternative form](https://docs.python.org/3/library/typing.html#typing.ParamSpec) we may be able to use here

I too would prefer that syntax, but AFAIK it's only supported in Python 3.12+. Based on your test matrix, it looks like you're targeting Python 3.8+, so unfortunately we need to use the old syntax for backwards compatibility. If you look at that [same page but for Python 3.11](https://docs.python.org/3.11/library/typing.html#typing.ParamSpec), you'll see that the example is different and matches what I have here.

> none of these changes seem relevant, we should revert

Sure thing 👍 @tobymao would you like me to make a separate PR for it?

Reply:
Ah I see, I didn't notice this was introduced in later versions.

Reply:
maybe? i'm not that interested in it if it's not 3.8+ compatable. but for sure let's remove it from this PR